src/Config/ObjectModel/McpRuntimeOptions.cs

```diff
@@ -51,7 +51,8 @@ public McpRuntimeOptions(
         // if DmlTools is null, set All tools enabled by default
         if (DmlTools is null)
         {
-            this.DmlTools = DmlToolsConfig.FromBoolean(DmlToolsConfig.DEFAULT_ENABLED);
+            // Use Default instead of FromBoolean to avoid setting UserProvided flags
+            this.DmlTools = DmlToolsConfig.Default;
```

Comment:
Where does FromBoolean get called from? Is there any reference? Do we still need that function?